cmd/charm/charmcmd/push_test.go

```diff
@@ -351,34 +351,26 @@ func (s *pushSuite) TestMapLogEntriesToVcsRevisions(c *gc.C) {
 func git(c *gc.C, tempDir string, arg ...string) {
 	cmd := exec.Command("git", arg...)
 	cmd.Dir = tempDir
-	output, err := cmd.Output()
-	errbuf := &bytes.Buffer{}
-	cmd.Stderr = errbuf
-	if err != nil {
-		c.Logf("result of git %v:%v with err: %v and stderr: %v", arg, output, err, errbuf.Bytes())
-	}
+	_, err := cmd.Output()
 	c.Assert(err, gc.IsNil)
 }
+
+// TODO frankban: really test the process of pushing VCS extra information,
+// including error cases. There is no need to export internal implementation
+// functions. Moreover, Bazaar tests seem to be missing.
+
 func (s *pushSuite) TestUpdateExtraInfoGit(c *gc.C) {
-	// Test suite deletes PATH, so add path for test git
-	// OSX may use brew to get git. Keep /usr/local/bin in this PATH.
-	os.Setenv("PATH", "/usr/bin:/usr/local/bin")
-	defer os.Setenv("PATH", "")
-	tempDir, err := ioutil.TempDir("", "charmcmd-push-test")
-	defer func() {
-		cmd := exec.Command("/bin/rm", "-r", tempDir)
-		err := cmd.Run()
-		c.Assert(err, gc.IsNil)
-	}()
-	c.Assert(err, gc.IsNil)
+	// Under OSX git is placed in /usr/local/bin by HomeBrew
+	s.PatchEnvironment("PATH", "/usr/bin:/usr/local/bin")
+	tempDir := c.MkDir()
 	git(c, tempDir, "init")
-	{
-		foo, err := os.Create(tempDir + "/foo")
-		c.Assert(err, gc.IsNil)
-		defer foo.Close()
-		_, err = foo.WriteString("bar")
-		c.Assert(err, gc.IsNil)
-	}
+
+	foo, err := os.Create(tempDir + "/foo")
+	c.Assert(err, gc.IsNil)
+	defer foo.Close()
```

Comment:
I think the old code was perhaps correct to close foo before
doing the git stuff, as under some platforms (e.g. Windows)
you won't be able to read the file while it's still open.

Why not just use ioutil.WriteFile instead?